src/vm/eetwain.cpp

```diff
@@ -3805,6 +3806,28 @@ bool UnwindEbpDoubleAlignFrame(
     {
         TADDR baseSP;
 
+#ifdef WIN64EXCEPTIONS
+        // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+        // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+        // If funclet frame layout is changed from CodeGen::genFuncletProlog() and genFuncletEpilog(),
+        // we need to change here accordingly.
+        if (pCodeInfo->IsFunclet())
+        {
+            baseSP = curESP + 8; // padding for 16byte stack alignment allocated in genFuncletProlog()
```

Comment:
We are trying to solve various issues given the current state of how we decided to do certain things on the fly. Given the knowledge of issues we have accumulated recently, I believe we should step back and reconsider the possibilities we have from higher level point of view. Some of the issues are due to the way we have decided to use for stack aligning. Some might be solved if we changed calling convention of jitted code so that the callee is no longer responsible for reclaiming the stack arguments space.
I am afraid that otherwise we would get lost in solving local issues that we could side step using a  different global approach.